### PR TITLE
Daily/1.0.0 make host and port passable, fetch from env for Docker run.

### PR DESCRIPTION
**Pull request by charlescui**
make host and port passable, fetch from env for Docker run. 

in the bash shell, before exec `npm run build`, make the env first:
```
export ZIPKIN_HOST=172.16.10.56 ZIPKIN_PORT=32008
```

in the dockerfile,
```
ENV ZIPKIN_HOST 172.16.10.56
ENV ZIPKIN_PORT 32008
COPY ./zipkin-ui /opt/ongo360-demo/
RUN cd /opt/ongo360-demo/zipkin-ui && \
    npm install typescript -g
    rm -rf ./node_modules && \
    npm install && \
    npm run build
```

Update feature for this issue #25

### CODE REVIEW

**Comment by adriancole** [comment]
can you rebase this on latest master? the changes to fix the build have already been fixed

File: src/app/tracegraph/tracegraph.component.ts [hunk 1]
```diff
@@ -9,15 +9,19 @@ import { Http } from '@angular/http';
...
+        this.baseUri = process.env.ZIPKIN_HOST || 'localhost';
+        this.basePort = process.env.ZIPKIN_PORT || '9411';
     }
 
     ngOnInit() {
 
         this
             .http
-            .get(`http://localhost:9411/api/v1/dependencies?endTs=1474206961061&lookback=86400000`, {})
+            .get(`http://${this.baseUri}:${this.basePort}/api/v1/dependencies?endTs=1474206961061&lookback=86400000`, {})
```
Comment (adriancole) [inline src/app/tracegraph/tracegraph.component.ts]: I wonder if this makes more sense as a base url than split into host/port?

**Comment by charlescui** [comment]
@adriancole OK, I give a PR #35

File: src/app/tracegraph/tracegraph.component.ts (hunk 1 above)
Comment (charlescui) [inline src/app/tracegraph/tracegraph.component.ts]: this.baseUri and this.basePort is OK, use only baseUri is not necessary.

Comment (ankon) [inline src/app/tracegraph/tracegraph.component.ts]: baseUri here is extremely misleading: a URI can already contain a port, so when reading this code I really would get confused. Given that you name the environment variables ZIPKIN_HOST and ZIPKIN_PORT, why rename them here?

**Comment by ankon** [comment]
A general note regarding the use of HOST and PORT separately: This immediately prevents using proxies that also add a path, and it also leaves it impossible to change the protocol now -- in my case for example it should be _https_ not _http_.

I think it would be easiest to capture all of these requirements by just requring a "base URL" to be provided - default would be 'http://localhost:9411'. ngOnInit() can then use proper URL/URI functions for manipulation, for example using urijs you'd end up with `URI('api/v1/dependencies').absoluteTo(this.baseUri).query({endTs: 1474206961061, lookback: 86400000'})`. No string manipulation, and reasonable flexible for operators to set up their environment.

**Comment by adriancole** [comment]
I agree with this logic about base URL. it is easier to deal with and
requires less parameters to work in scenarios like proxies. Mind giving it
a go? would be nice to have this working.

**Comment by adriancole** [comment]
can someone update this impl to use ZIPKIN_BASE_URL? per https://github.com/openzipkin/docker-zipkin/issues/134#issuecomment-288703867

**Comment by llinder** [comment]
I will take a look.  I've been meaning to try out the new UI anyway so this will give me an excuse to build and run it locally.

**Comment by llinder** [comment]
Rebased this PR against master and changed to a single environment variable 'ZIPKIN_BASE_URL'  https://github.com/openzipkin/zipkin-ui/pull/37

**Comment by adriancole** [comment]
Thank you!

**Comment by adriancole** [comment]
closed via #37

**Pull request closed without merging**